notebooks/03-query/09-functions-case.ipynb

```diff
@@ -0,0 +1,499 @@
+    "\n",
+    "This notebook uses the SQL versions of `CASE`. Native (JSON-based) versions [are also available](https://druid.apache.org/docs/27.0.0/querying/math-expr#general-functions), allowing them to be used in streaming ingestion, for example.\n",
+    "\n",
+    "### Simple CASE\n",
+    "\n",
+    "The following SQL shows a simple CASE statement being used to tweak the results of a query based on the browser that someone was using on the site.\n",
+    "\n",
+    "The first parameter for the `CASE` is the dimension, and then each `WHEN` is followed by a specific value to compare with."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "984358e6-4895-49a9-8f32-1baa14c7123d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  browser,\n",
+    "  AVG(\"session_length\") AS \"average_session_length\",\n",
+    "  AVG(\n",
+    "      CASE \"browser\"\n",
+    "          WHEN 'IE' THEN session_length * 2\n",
+    "          WHEN 'Chrome' THEN session_length / 2\n",
+    "          ELSE session_length\n",
+    "          END\n",
+    "          ) AS \"average_session_length_maybe\"\n",
+    "FROM \"example-koalas-conditions\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'2019-08-25T0/PT1H')\n",
+    "GROUP BY 1\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "29bb64c9-9c4c-476b-9718-51c1c35a3d05",
+   "metadata": {},
+   "source": [
+    "### Searched CASE\n",
+    "\n",
+    "The second form of `CASE` is not pinned to a specific dimension or expression, allowing for more complex conditions.\n",
+    "\n",
+    "A common use of `CASE` is to tag particular events as significant.\n",
+    "\n",
+    "With some artistic license, we can apply this to the example dataset. The SQL statement below flags some events as impressions (when someone first saw an image) and others as conversions (when someone completed the task and finished clearing the image)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e8f68a70-e629-4ba7-a28d-5ba282d8469b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  CASE WHEN (\"event_type\" = 'GoodLoad') THEN 'yes' ELSE 'no' END AS \"isImpression\",\n",
+    "  CASE WHEN (\"event_subtype\" IS NULL AND \"event_type\" = 'LayerClear') THEN 'yes' ELSE 'no' END AS \"isConversion\",\n",
+    "  COUNT(*) AS \"Count\"\n",
+    "FROM \"example-koalas-conditions\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'2019-08-25T0/PT4H')\n",
+    "GROUP BY 1, 2\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b266e5bb-a4b8-4976-94b4-e94d8d257b44",
+   "metadata": {},
+   "source": [
+    "Combining `CASE` with `SUM` leads to an interesting pattern: the SQL below shows an approach to finding the number of conversions per image shown."
```

Comment:
Maybe include a brief description on what the regex is doing?

Reply:
Done.